wlm/src/main/java/ai/djl/serving/wlm/ModelInfo.java

```diff
@@ -970,32 +976,37 @@ void downloadS3() throws ModelException, IOException {
             return;
         }
         if (modelId.startsWith("s3://")) {
-            logger.info("{}: S3 url found, start downloading from {}", uid, modelId);
-            // Use fixed download path to avoid repeat download
-            String hash = Utils.hash(modelId);
-            String download = Utils.getenv("SERVING_DOWNLOAD_DIR", null);
-            Path parent = download == null ? Utils.getCacheDir() : Paths.get(download);
-            parent = parent.resolve("download");
-            this.downloadDir = parent.resolve(hash);
-            if (Files.exists(this.downloadDir)) {
-                logger.info("{}: artifacts has been downloaded already: {}", uid, this.downloadDir);
-                return;
-            }
+            this.downloadDir = downloadS3(modelId);
+        } else if (modelId.startsWith("djl://")) {
+            logger.info("{}: djl model zoo url found: {}", uid, modelId);
+            modelUrl = modelId;
+            // download real model from model zoo
+            downloadModel();
+        }
+    }
+
+    private Path downloadS3(String s3Url) throws IOException, ModelException {
```

Comment:
Better give a different name, method overload requires the same return value. This is not a overload case. consider something like: downloadS3Internal